### ISSUE TITLE
Bug: Popover components interfere with keyboard navigation on <a> elements contained within <PopoverButton> components

### ISSUE THREAD

**Issue by sps-nathan-alden-sr** [tailwindlabs/headlessui#1152]
**What package within Headless UI are you using?**

`@headlessui/vue`

**What version of that package are you using?**

1.5.0

**What browser are you using?**

Firefox latest
Chrome latest

**Reproduction URL**

https://github.com/sps-nathan-alden-sr/tailwindui-popover-bug

Run `yarn`, then `yarn serve` to start the app.

**Describe your issue**

It seems that popover components interfere with the proper function of rendered `<a>` elements that are contained within `<PopoverButton>` components. In the GIF below, you can see how I tab to the links and press `ENTER`, which is detected by the event handler, but no navigation actually occurs. Clicking the links works as expected, as demonstrated at the very end of the animation.

![navigation](https://user-images.githubusercontent.com/86692513/155584014-364be487-1552-4048-8ca2-3654b3cd584b.gif)

The component in question is <https://github.com/sps-nathan-alden-sr/tailwindui-popover-bug/blob/master/src/components/TopNavComponent.vue>.

**Comment by adamwathan** [comment]
Hey! Have 5 seconds so just responding quickly — this looks like a mistake in how you're using it to me, each link needs its own `<PopoverButton>` and you need to make sure the button is rendering as your link:

```html
<PopoverGroup>
  <Popover>
    <PopoverButton>Popover</PopoverButton>
    <PopoverPanel style="margin-left: 10px">
      <PopoverButton as="template">
        <router-link :to="{ name: 'Home' }">Home</router-link>
      </PopoverButton>
      <br />
      <PopoverButton as="template">
        <router-link :to="{ name: 'About' }">About</router-link>
      </PopoverButton>
    </PopoverPanel>
  </Popover>
</PopoverGroup>
```

**Comment by sps-nathan-alden-sr** [comment]
Unfortunately, that did not resolve the problem. My new markup looks like this:
```html
<PopoverGroup>
  <Popover>
    <PopoverButton>Popover</PopoverButton>
    <PopoverPanel style="margin-left: 10px;">
      <PopoverButton as="template">
        <router-link :to="{ name: 'Home' }" @keydown="k">Home</router-link>
      </PopoverButton>
      <br />
      <PopoverButton as="template">
        <router-link :to="{ name: 'About' }" @keydown="k">About</router-link>
      </PopoverButton>
    </PopoverPanel>
  </Popover>
</PopoverGroup>
```
I was able to get rid of the `tabindex="-1"` after adding `as="template"` but the links still do not navigate. The `keydown` handler is still fired, however.

**Comment by adamwathan** [comment]
Had a chance to clone it down and yep you are totally right, doesn't work! 🤔 It doesn't work with a vanilla `a` tag for me either. Will investigate.

**Comment by sps-nathan-alden-sr** [comment]
Yeah, I tried regular `<a>` tags, as well, thinking it might've been `<router-link>` magic.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

I was also to reproduce it locally in a test and wrote a fix for it. I tested it on your repo and it works with normal `a` and `router-link` components.

This should be fixed by #1153, and will be available in the next release.

You can already try it using `npm install @headlessui/vue@insiders` or `yarn add @headlessui/vue@insiders`.
